lua/neoclip/utils.lua

```diff
@@ -11,6 +11,20 @@ M.join = function(t1, t2)
     return new_t
 end
 
+--- Iterates over the given table, calling the given function on all items. The results
+--- of the function will be used to create a new table.
+---
+--- @generic Item
+--- @param tbl table<number, Item>
+--- @param fun fun(item: Item): Item
+M.map_table = function (tbl, fun)
```

Comment:
Could instead use `vim.tbl_map`?

Reply:
I had no idea this existed! It'll make my life a lot easier :P